### PR TITLE
ci: ensure installing docker compose 2.18.1 and re-enable the integration tests

### PR DESCRIPTION
**Pull request by eitsupi**
Continuing from #2759 and #2764

### CODE REVIEW

File: .github/workflows/test-rust.yaml [hunk 1]
```diff
@@ -54,25 +54,22 @@ jobs:
...
-      #     resource: "tcp:5432 tcp:3306 tcp:1433"
-      #     timeout: 60000
-      #   if: ${{ inputs.os == 'ubuntu-latest' }}
+      - name: Run docker compose
+        # docker compose v2.18.0 breaks; we can remove when the latest version on the runners is v2.18.1
+        run: |
+          export DOCKER_CONFIG="${PWD}/.docker"
+          mkdir -p "${DOCKER_CONFIG}/cli-plugins"
+          curl -SL https://github.com/docker/compose/releases/download/v2.18.1/docker-compose-linux-x86_64 -o "${DOCKER_CONFIG}/cli-plugins/docker-compose"
+          chmod +x "${DOCKER_CONFIG}/cli-plugins/docker-compose"
```
Comment (eitsupi) [inline .github/workflows/test-rust.yaml:63]:
```suggestion
          chmod +x "${DOCKER_CONFIG}/cli-plugins/docker-compose"
          docker compose --version
```

File: .github/workflows/test-rust.yaml [hunk 2]
```diff
@@ -54,25 +54,23 @@ jobs:
...
+          docker compose --version
```
Comment (eitsupi) [inline .github/workflows/test-rust.yaml]:
```suggestion
          docker compose version
```

**Comment by max-sixty** [comment]
Looks good if it works!!

**Comment by eitsupi** [comment]
It seems working ✌️

**Comment by max-sixty** [comment]
Nice job!

**Pull request merged**